### PR TITLE
Card Design Alignment

### PR DESCRIPTION
**Pull request by jwijay**
#### What does this PR do?
Re-aligning styles with HPE toolkit Card designs.

#### What testing has been done on this PR?
Tested grommet PR branch against hpe-digitaltoolkit and grommet-docs branches in Chrome, Safari, and Firefox.

#### What are the relevant issues?
https://github.com/grommet/hpe-digitaltoolkit/issues/31
https://github.com/grommet/hpe-digitaltoolkit/issues/94

#### Screenshots (if appropriate)
**_Box 1-4 Basis + Pad-Between Wrapping Issue (before PR):_**
box-basis-1-4-wrapping

**_Box 1-4 Basis + Pad-Between Wrapping Fix (after PR):_**
columns-basis-padding-between-fix

_**Card Horizontal Mobile Video Thumbnail Too Short + Watch Icon Small (before PR):**_
card-horizontal-mobile-video-thumbnail-short

_**Card Horizontal Mobile Video Thumbnail Height + Watch Icon Size Fix (after PR):**_
card-mobile-watch-icon-fix

_**Card Tiles Background Short (before PR):**_
![card-tiles-bg-short]
(https://cloud.githubusercontent.com/assets/10161095/18461360/8562b90c-7914-11e6-9d26-47e8e7a6df58.png)

_**Card Tiles Background Height Fix (after PR):**_
card-tiles-bg-short-fix

_**Card Mobile Width Too Narrow + Margin (before PR):**_
mobile-margin

_**Card Mobile Width + Margin Fix (after PR):**_
mobile-margin-fix

_**Columns Flex-Basis: Auto Default (before PR):**_
columns-flex-basis-auto-default-bug

_**Columns Flex-Basis Medium Size Default (after PR):**_
columns-fix

### CODE REVIEW

File: src/scss/grommet-core/_objects.card.scss [hunk 1]
```diff
@@ -39,7 +39,14 @@ $card-hover-color: #EBEBEB;
     }
   }
 
-  // &__thumbnail {
+  &.#{$grommet-namespace}box--responsive {
```
Comment (ericsoderberghp) [inline src/scss/grommet-core/_objects.card.scss]: I would have thought we got this behavior automatically from box. How is this changing it?

File: src/scss/grommet-core/_objects.card.scss [hunk 2]
```diff
@@ -50,14 +57,14 @@ $card-hover-color: #EBEBEB;
...
   //
-  //   .#{$grommet-namespace}control-icon-watch {
-  //     @include media-query(palm) {
-  //       // don't shrink video WatchIcon when in mobile
-  //       width: round($inuit-base-spacing-unit * 6);
-  //       height: round($inuit-base-spacing-unit * 6);
-  //     }
-  //   }
-  // }
+    .#{$grommet-namespace}control-icon-watch {
```
Comment (ericsoderberghp) [inline src/scss/grommet-core/_objects.card.scss]: Can we add this to the video icon in general? Perhaps a `responsive={false}` on the icons?

File: src/scss/grommet-core/_objects.card.scss (hunk 1 above)
Comment (jwijay) [inline src/scss/grommet-core/_objects.card.scss]: .grommetux-box--width-medium, and margins persist in mobile:
mobile-margin

This section overrides the width and margin:
mobile-margin-fix

File: src/scss/grommet-core/_objects.card.scss (hunk 2 above)
Comment (jwijay) [inline src/scss/grommet-core/_objects.card.scss]: @ericsoderberghp 
Added responsive option to icons in grommet-icon-loader: https://github.com/grommet/grommet-icon-loader/pull/4

Also removed WatchIcon mobile styles from Card styles, and updated Icon styles:
https://github.com/grommet/grommet/pull/852/commits/97f131d20cb717cd56f81d609dc6572c77136192#diff-d03f8eb3b97425583c39fc4f55957b12R72

**Pull request merged**